src/main/java/edu/unc/lib/boxc/migration/cdm/services/CdmFieldService.java

```diff
@@ -0,0 +1,231 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.boxc.migration.cdm.services;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import edu.unc.lib.boxc.common.util.URIUtil;
+import edu.unc.lib.boxc.migration.cdm.exceptions.InvalidProjectStateException;
+import edu.unc.lib.boxc.migration.cdm.exceptions.MigrationException;
+import edu.unc.lib.boxc.migration.cdm.model.CdmFieldInfo;
+import edu.unc.lib.boxc.migration.cdm.model.CdmFieldInfo.CdmFieldEntry;
+import edu.unc.lib.boxc.migration.cdm.model.MigrationProject;
+
+/**
+ * Service for persistence and retrieval of CDM field information
+ *
+ * @author bbpennel
+ */
+public class CdmFieldService {
+    private CloseableHttpClient httpClient;
+    private String cdmBaseUri;
+
+    private static final String CDM_NICK_FIELD = "nick";
+    private static final String CDM_NAME_FIELD = "name";
+
+    public static final String EXPORT_NICK_FIELD = "cdm_nick";
+    public static final String EXPORT_AS_FIELD = "export_as";
+    public static final String EXPORT_DESC_FIELD = "description";
+    public static final String EXPORT_SKIP_FIELD = "skip_export";
+    public static final String[] EXPORT_CSV_HEADERS = new String[] {
+            EXPORT_NICK_FIELD, EXPORT_AS_FIELD, EXPORT_DESC_FIELD, EXPORT_SKIP_FIELD };
+
+    public CdmFieldService() {
+    }
+
+    /**
+     * Get the URL for retrieving field info for the given collection
+     * @param cdmBaseUri
+     * @param collectionId
+     * @return
+     */
+    public static String getFieldInfoUrl(String collectionId) {
+        return "dmwebservices/index.php?q=dmGetCollectionFieldInfo/" + collectionId + "/json";
```

Comment:
Don't know if things ever improved on the CDM side, but getting JSON sometimes returned a less complete response than getting XML.

Reply:
The responses seem to be complete for getting collection field info, and it is the same approach being used in the NCDHC migration scripts, so I'm guessing it will be consistent. But i did see the same issue you're talking about when it comes to getting item records as json, it would only give me up to 5 fields.